### ISSUE TITLE
naming things

### ISSUE THREAD

**Issue by cboettig** [cboettig/contentid#34]
Names are so hard.

- `history`. I'm still happy about having `query()` divided into separate functions for getting the `history()` of a URL vs getting the `sources()` of a content identifier, but `history()` is a namespace collision with `utils::history`.  I'm currently thinking `url_history()`?  (I know neither of these are verbs... suggestions welcome -- the only verb that comes to mind is something like `get_history()`... maybe that's better?).

- Maybe `url_history()` suggests that `sources()` should be `content_sources()`? Or we go for `get_sources()` and `get_history()`?

- Package name (again).  I'm thinking of calling it `contentid` instead of `contenturi`.  I don't think the more precise term URI means something to most researchers, (and at some level the core concept is not URI-specific).  Also, `contentid` has a pleasant double entendre.  

I will probably fret about more names later so figured I'd collect that in one thread...

**Comment by jhpoelen** [comment]
I interpret the ```sources``` / ```history``` as short cuts to fill-in-the-blank queries:

```
_:blank <hasVersion> <hash://sha256/1234...> .
```
and 

```
<https://example.org/data.zip> <hasVersion> _:blank .
```

so, the ```query_``` prefix makes sense to me.

**Comment by cboettig** [comment]
yes!  `query_history()` and `query_sources()` it is then.  and we've adopted `contentid`.  feeling better about most of our namesspace for now.

poking around in for the identifiers discussion I did stumble across the fact that `pinning` is also a concept in IPFS, which is perhaps also a useful verb for our content store bit, though has a rather different meaning than `pins::pin()`.  But at least for the moment I see our `contentid::pin()` as more of a proof of concept than a core verb anyway, `register()` <--> `query_*`, `store()` <--> `retrieve()` make for a nice compact set.   `resolve()` provides a higher-level function with a thin bit of automation around `query_sources()`, and `content_id()`  provides a simple low-level function for checking a hash of remote or local stuff without registering it.
